vision/standard_object/odlc_classify_shape.py

```diff
@@ -69,420 +89,339 @@ def process_shapes(
-    )
-
-    return len(defects) if defects is not None else 0
-
-
-def get_angle(
-    pts: tuple[
-        npt.NDArray[npt.Shape["1, 2"], npt.IntC],
-        npt.NDArray[npt.Shape["1, 2"], npt.IntC],
-        npt.NDArray[npt.Shape["1, 2"], npt.IntC],
-    ]
-) -> npt.Float64:
+def generate_polar_array(cnt: consts.Contour) -> Tuple[List[float], List[float]]:
+    # def generate_polar_array(cnt: consts.Contour) -> chars.ODLCShape | None:
     """
-    Takes 3 points and calculates the angle they make.
+    Generates 2 arrays storing the x and y coordinates of a new polar array
 
     Parameters
     ----------
-    pts : tuple[
-        npt.NDArray[npt.Shape["1, 2"], npt.IntC],
-        npt.NDArray[npt.Shape["1, 2"], npt.IntC],
-        npt.NDArray[npt.Shape["1, 2"], npt.IntC]
-    ]
-        Three points that make an angle that will be calculated
-        All points formatted as done in openCV [[y, x]].
-        pt_a : npt.NDArray[npt.Shape["1, 2"], npt.IntC]
-            One of the points that form the angle, 0th index in the tuple.
-        vertex : npt.NDArray[npt.Shape["1, 2"], npt.IntC]
-            The point that is in the middle, where the angle is, 1st index in the tuple.
-        pt_b : npt.NDArray[npt.Shape["1, 2"], npt.IntC]
-            The other point that forms the angle, 2nd index in the tuple.
+    cnt : consts.Contour
+        A contour returned by the contour detection algorithm
 
     Returns
     -------
-    angle : npt.Float64
-        The angle (in deg) of the angle formed by the lines (pt_a, vertex) and (vertex, pt_b)
-
-    Notes
-    -----
-    Angle formula is derived from vector dot product (the one that uses the angle between vectors)
-    If there are vectors a and b with angle t between them then:
-        a.b = ||a||*||b||*cos(t)
-        (Where a.b is the dot product of a and b)
-    This can be rearranged to:
-        t = arccos((a.b) / (||a||*||b||))
+    shape : chars.ODLCShape | None
+        Will return one of the ODLC shapes defined in vision/common/odlc_characteristics or None
+        if the given contour is not an ODLC shape (doesnt match any ODLC)
     """
-    # vector forms of points a (pts[0]) and b (pts[2]) with vertex (pts[1]) as origin
-    vec_a: npt.NDArray[npt.Shape["1, 2"], npt.IntC] = pts[0] - pts[1]
-    vec_b: npt.NDArray[npt.Shape["1, 2"], npt.IntC] = pts[2] - pts[1]
 
-    vec_a = np.squeeze(vec_a)
-    vec_b = np.squeeze(vec_b)
-
-    return np.degrees(
-        np.arccos(np.dot(vec_a, vec_b) / (np.linalg.norm(vec_a) * np.linalg.norm(vec_b)))
-    )
-
-
-def get_angles(approx: consts.Contour) -> npt.NDArray[npt.Shape["*"], npt.Float64]:
+    x_avg: Float64 = 0
+    y_avg: Float64 = 0
+    numPoints: int = 0
+    point: NDArray[Shape["2, 2"], IntC, Float64]
+    # Finds average x and y value
+    for point in cnt:
+        y_avg += point[0][0]
+        x_avg += point[0][1]
+        numPoints += 1
+
+    x_avg = x_avg // numPoints
+    y_avg = y_avg // numPoints
```

Comment:
You can replace this calculation of the average with `x_avg, y_avg = np.mean(cnt, axis=2)`. I'm not 100% sure that the axis is 2, but something similar should work